dex/msgjson/types.go

```diff
@@ -644,12 +645,14 @@ type OrderBook struct {
 
 // MatchProofNote is the match_proof notification payload.
 type MatchProofNote struct {
-	MarketID  string  `json:"marketid"`
-	Epoch     uint64  `json:"epoch"`
-	Preimages []Bytes `json:"preimages"`
-	Misses    []Bytes `json:"misses"`
-	CSum      Bytes   `json:"csum"`
-	Seed      Bytes   `json:"seed"`
+	MarketID string `json:"marketid"`
+	Epoch    uint64 `json:"epoch"`
+	// Note: ppting to use order.Preimage and order.OrderID types
+	// here to avoid casting client side.
+	Preimages []order.Preimage `json:"preimages"`
+	Misses    []order.OrderID  `json:"misses"`
```

Comment:
I'm uneasy about this because it adds a dependency to msgjson, and it goes against the approach used in all the other message types.  Also, there is no TestMatchProofNote in types_test.go (that's my fault).

Reply:
But at the heart of it, it violates the spec as it marshals to something else entirely.

e.g.

`{"marketid":"adsf","epoch":1234,"preimages":[[1,123,255,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]],"misses":[[42,92,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]],"csum":"03","seed":"03"}`

vs.

`{"marketid":"adsf","epoch":1234,"preimages":["017bff"],"misses":["2a5c01"],"csum":"03","seed":"03"}`